pkg/agent/multicluster/mc_route_controller.go

```diff
@@ -214,39 +240,192 @@ func (c *MCRouteController) Run(stopCh <-chan struct{}) {
+		if !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to get Gateway %s, error: %s", gateway, err.Error())
+		}
+		klog.InfoS("Gateway is deleted, removing WireGuard interface and rule")
+		if err := c.deleteHostRoute(); err != nil {
+			return err
+		}
+		c.wireGuardInitialized = false
+	}
+
+	return nil
+}
+
+func (c *MCRouteController) deleteHostRoute() error {
+	if err := c.routeClient.DeleteIPRule(int(openflow.MulticlusterPktMark), multiclusterWireGuardTable); err != nil {
+		return err
+	}
+	link, err := c.routeClient.LinkByName(multiclusterWireGuardInterface)
+	if err != nil {
+		return nil
+	}
+	if err := c.routeClient.DeleteLink(link); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *MCRouteController) initHostRoute() error {
+	if err := c.routeClient.AddIPRule(int(openflow.MulticlusterPktMark), multiclusterWireGuardTable); err != nil {
+		return err
+	}
+	link, err := c.routeClient.LinkByName(multiclusterWireGuardInterface)
+	if err != nil {
+		return err
+	}
+	if err := c.routeClient.AddMulticlusterRoute(link, net.IPv4zero, multiclusterWireGuardTable); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *MCRouteController) updatePeers(ciImp string) error {
+	ciImport, err := c.mcClient.MulticlusterV1alpha1().ClusterInfoImports(c.namespace).Get(context.TODO(), ciImp, metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+		klog.InfoS("The ClusterInfoImport has been deleted, removing peer", "ClusterInfoImport", ciImp)
+		return c.wireGuardClient.DeletePeer(ciImp)
+	}
+	if !c.wireGuardInitialized {
+		return fmt.Errorf("failed to update WireGuard peers, the WireGuard has not been initialized")
+	}
+	gatewayIP := net.ParseIP(ciImport.Spec.GatewayInfos[0].GatewayIP)
+	_, gatewayIPCIDR, err := net.ParseCIDR(ciImport.Spec.GatewayInfos[0].GatewayIP + "/32")
+	if err != nil {
+		return err
+	}
+	_, serviceIPCIDR, err := net.ParseCIDR(ciImport.Spec.ServiceCIDR)
+	if err != nil {
+		return err
+	}
+	podCIDRs := []*net.IPNet{}
+	for _, cidr := range ciImport.Spec.PodCIDRs {
+		_, podCIDR, err := net.ParseCIDR(cidr)
+		if err != nil {
+			return err
+		}
+		podCIDRs = append(podCIDRs, podCIDR)
+	}
+
+	allowedIPs := []*net.IPNet{serviceIPCIDR, gatewayIPCIDR}
+	allowedIPs = append(allowedIPs, podCIDRs...)
+	if err := c.wireGuardClient.UpdatePeer(ciImport.Name, ciImport.Spec.WireGuard.PublicKey, gatewayIP, ciImport.Spec.WireGuard.Port, allowedIPs); err != nil {
+		return err
+	}
+	klog.InfoS("Update WireGuard peer with ClusterInfoImport", "ClusterInfoImport", ciImp)
+	return nil
+}
+
+func (c *MCRouteController) initializeWireGuardInterface() error {
+	c.wireGuardConfig.MTU = c.nodeConfig.NodeMTU - config.WireGuardOverhead - config.GeneveOverhead
+	wgClient, err := wireGuardNewFunc(nil, c.mcClient, c.nodeConfig, c.wireGuardConfig, c.namespace)
+	if err != nil {
+		return err
+	}
+	c.wireGuardClient = wgClient
```

Comment:
Is it necessary to call client initialization for every event? I feel this part can be moved into `NewMCRouteController`